server.js

```
@@ -0,0 +1,218 @@
+          } else {
+            directory = req.url
+            contentType = req.url.slice(req.url.indexOf('.') + 1, req.url.length);
+          }
+        }
+        fs.readFile(`./public${directory}`, (err, data) => { // Sends requested data
+          if (err) {
+            return console.log(err);
+          }
+          res.writeHead(200, {
+            "Content-Type": `text/${contentType}, charset=utf-8`,
+            "Content-Length": data.toString().length,
+            Date: date
+          });
+          res.write(data.toString());
+          res.end();
+        });
+      });
+    }
+    /*** POST Method ***/
+    if (req.method === "POST") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      // Used for new element pages
+      addOrUpdateElem(decodeChunk, req, res)
+      // Update index.html
+      parseNewHTML(decodeChunk, req, res);
+      res.writeHead(200, {
+        "Content-Type": 'application/json',
+        "Success": true,
+        "Date": date
+      })
+      res.end();
+    }
+    /*** PUT Method ***/
+    if (req.method === "PUT") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      fs.readdir("./public/", (err, files) => {
+        if (err) {
+          console.log(err)
+        }
+        if (files.includes(req.url.slice(1)) === true) {
+          addOrUpdateElem(decodeChunk, req, res)
+        } else {
+          res.writeHead(500, {
+            "Content-Type": `application/json, charset=utf-8`,
+            "error": `resource ${req.url} does not exist`,
+            "Date": date
+          });
+          res.end();
+        }
+      });
+    }
+    /*** DELETE Method ***/
+    if (req.method === "DELETE") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      fs.readdir("./public/", (err, files) => {
+        if (err) {
+          return console.log(err);
```

This should return an error to the client using either a 400/500 error